src/EnergyPlus/PollutionModule.cc

```diff
@@ -186,1983 +186,223 @@ void GetPollutionFactorInput(EnergyPlusData &state)
-            if (!state.dataIPShortCut->lAlphaFieldBlanks(14)) {
-                CheckFFSchedule(state,
-                                cCurrentModuleObject,
-                                "OtherFuel2",
-                                state.dataIPShortCut->cAlphaFieldNames(14),
-                                state.dataIPShortCut->cAlphaArgs(14),
-                                Pollution.OtherFuel2Coef.HgSched,
-                                ErrorsFound);
-            }
-            Pollution.OtherFuel2Coef.Pb = state.dataIPShortCut->rNumericArgs(14);
-            if (!state.dataIPShortCut->lAlphaFieldBlanks(15)) {
-                CheckFFSchedule(state,
-                                cCurrentModuleObject,
-                                "OtherFuel2",
-                                state.dataIPShortCut->cAlphaFieldNames(15),
-                                state.dataIPShortCut->cAlphaArgs(15),
-                                Pollution.OtherFuel2Coef.PbSched,
-                                ErrorsFound);
-            }
-            Pollution.OtherFuel2Coef.Water = state.dataIPShortCut->rNumericArgs(15);
-            if (!state.dataIPShortCut->lAlphaFieldBlanks(16)) {
-                CheckFFSchedule(state,
-                                cCurrentModuleObject,
-                                "OtherFuel2",
-                                state.dataIPShortCut->cAlphaFieldNames(16),
-                                state.dataIPShortCut->cAlphaArgs(16),
-                                Pollution.OtherFuel2Coef.WaterSched,
-                                ErrorsFound);
-            }
-            Pollution.OtherFuel2Coef.NucHi = state.dataIPShortCut->rNumericArgs(16);
-            if (!state.dataIPShortCut->lAlphaFieldBlanks(17)) {
-                CheckFFSchedule(state,
-                                cCurrentModuleObject,
-                                "OtherFuel2",
-                                state.dataIPShortCut->cAlphaFieldNames(17),
-                                state.dataIPShortCut->cAlphaArgs(17),
-                                Pollution.OtherFuel2Coef.NucHiSched,
-                                ErrorsFound);
-            }
-            Pollution.OtherFuel2Coef.NucLo = state.dataIPShortCut->rNumericArgs(17);
-            if (!state.dataIPShortCut->lAlphaFieldBlanks(18)) {
-                CheckFFSchedule(state,
-                                cCurrentModuleObject,
-                                "OtherFuel2",
-                                state.dataIPShortCut->cAlphaFieldNames(18),
-                                state.dataIPShortCut->cAlphaArgs(18),
-                                Pollution.OtherFuel2Coef.NucLoSched,
-                                ErrorsFound);
-            }
-        } break;
-        default: {
-            ShowSevereError(state, format("Illegal FuelType for Pollution Calc Entered={}", FuelType.FuelTypeNames(Loop)));
+        ip->getObjectItem(state,
```

Comment:
This block of code replaces the entire copy-pasted mess above.

Reply:
:open_mouth: